### DISCUSSION TITLE
using `--scope` with `version`/`publish`

### DISCUSSION THREAD

**Discussion by farfromrefug** [ghiscoding/lerna-lite#946] [Q&A] [answered]
Is there any way to filter/scope packages while using `version`/publish`? Let say i am in independent mode and want to only publish one package

**Comment by ghiscoding** [comment]
Lerna never supported the `--scope` as you can see in these Lerna issues
- https://github.com/lerna/lerna/issues/3874
- https://github.com/lerna/lerna/issues/2915

the main reason is that the code implementation never use the [filter-package](https://github.com/lerna-lite/lerna-lite/tree/main/packages/filter-packages) package, and it's only used by lerna exec/list/run/watch. Since Lerna itself never supported it, then Lerna-Lite doesn't either since it's the same code. I also don't know how to do these kind of filter since I never actually used the `independent` mode, I would suggest to search in Lerna's issues and maybe you'll find something in there

**Reply by farfromrefug** [reply]
@ghiscoding thanks i understand. I am now trying to implement "filtering" (scope, ignore ...) for all commands in lerna-lite. I already have something to test. Is there an easy way to test lerna commands within the `lerna-lite` cloned repo?
I have run `npm run build` successfully but i would now like to test running command like `lerna version` with what i just built

**Reply by ghiscoding** [reply]
I added VSCode debugger that you can use with breakpoints

https://github.com/lerna-lite/lerna-lite/blob/37f5f9758417b779d2f8636949364df35be87305/.vscode/launch.json#L19-L28

you can also see the [Troubleshooting](https://github.com/lerna-lite/lerna-lite/wiki/Troubleshooting) wiki

**Reply by farfromrefug** [reply, marked as answer]
@ghiscoding thanks ! here is the PR https://github.com/lerna-lite/lerna-lite/pull/948
